styx-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java

```diff
@@ -169,6 +193,20 @@ WorkflowInstanceExecutionData executionData(WorkflowInstance workflowInstance) t
     return WorkflowInstanceExecutionData.fromEvents(events);
   }
 
+  private List<WorkflowInstanceExecutionData> executionData(
+      Set<WorkflowInstance> workflowInstancesSet) {
+    return workflowInstancesSet.parallelStream()
+        .map(workflowInstance -> {
+          try {
+            return executionData(workflowInstance);
+          } catch (IOException e) {
+            throw Throwables.propagate(e);
+          }
+        })
+        .sorted(WorkflowInstanceExecutionData.COMPARATOR)
```

Comment:
Isn't it because the input collection is a Set, that does not guarantee any order? It might be a SortedSet that is actually passed in here, but I would guess that the additional sorting here is to make sure that any Set that is passed in would actually be correctly sorted.